python/cuml/internals/array.py

```diff
@@ -1172,12 +1172,16 @@ def from_input(
         if (
             not fail_on_order and order != arr.order and order != "K"
         ) or make_copy:
```

Comment:
Since we are checking this within the `if`, we could drop this check

```suggestion
        ):
```

Reply:
I suspect you are right and this can be simplified.  But deleting it would not make a copy at all.
We could maybe delete the whole `if` (always creating a new `arr`) or assuming a copy is always made (I am not certain that is currently true for e.g. 1-D arrays).

Reply:
We do not always do a copy and not always want to. If you see above in the function, the `make_copy` variable is inferred from other conditions like 

```
make_copy = force_contiguous and not arr.is_contiguous
```

which would make the complex conditional worse if we wanted to pack everything in. So we do not want to delete this.

Reply:
Ok maybe there's a better way to write the code below then: https://github.com/rapidsai/cuml/pull/5954#discussion_r1681579685